### PR TITLE
docs: update to eslint v9 in custom-rule-tutorial

### PR DESCRIPTION
**Pull request by aladdin-add**
fixes https://github.com/eslint/eslint/issues/18382

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

### CODE REVIEW

File: docs/src/extend/custom-rule-tutorial.md [hunk 1]
```diff
@@ -217,7 +217,7 @@ const fooBarRule = require("./enforce-foo-bar");
 const ruleTester = new RuleTester({
   // Must use at least ecmaVersion 2015 because
   // that's when `const` variables were introduced.
-  parserOptions: { ecmaVersion: 2015 }
+  languageOptions: { ecmaVersion: 2015 }
```
Comment (mdjermanovic) [inline docs/src/extend/custom-rule-tutorial.md:220]: We should update the same in the example project that is linked from the tutorial:

https://github.com/eslint/eslint/blob/b4d2512809a1b28466ad1ce5af9d01c181b9bf9e/docs/_examples/custom-rule-tutorial-code/enforce-foo-bar.test.js#L10-L14

Comment (mdjermanovic) [reply docs/src/extend/custom-rule-tutorial.md:220]: We should probably also update all references to ESLint v8 (e.g, devDependencies, peerDependencies) to ESLint v9.

Comment (aladdin-add) [reply docs/src/extend/custom-rule-tutorial.md:220]: 005ee25f29998c8012d1ed95b62f04b85da4d543

File: docs/_examples/custom-rule-tutorial-code/package.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "name": "eslint-plugin-example",
+  "private": true,
```
Comment (aladdin-add) [inline docs/_examples/custom-rule-tutorial-code/package.json:3]: I also added this property - to avoid it being published unexpectedly.

**Comment by mdjermanovic** [comment]
Few more places to update to v9:

https://github.com/eslint/eslint/blob/005ee25f29998c8012d1ed95b62f04b85da4d543/docs/src/extend/custom-rule-tutorial.md?plain=1#L365

https://github.com/eslint/eslint/blob/005ee25f29998c8012d1ed95b62f04b85da4d543/docs/src/extend/custom-rule-tutorial.md?plain=1#L382-L385

https://github.com/eslint/eslint/blob/005ee25f29998c8012d1ed95b62f04b85da4d543/docs/src/extend/custom-rule-tutorial.md?plain=1#L394-L396

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**